app/store/middleware/onChangedSelectedHealthAuthorities.ts

```diff
@@ -1,22 +1,37 @@
 import { Dispatch, Middleware, AnyAction, MiddlewareAPI } from 'redux';
 import IntersectService from '../../services/IntersectService';
+import toggleSelectedHealthcareAuthorityAction from '../actions/healthcareAuthorities/toggleSelectedHealthcareAuthorityAction';
+import { HealthcareAuthority } from '../../api/healthcareAuthorities/getHealthcareAuthoritiesApi';
+
+const areHaListsEqual = (
+  listA: HealthcareAuthority[],
+  listB: HealthcareAuthority[],
+) => {
+  const haIdSetA = new Set(listA.map((ha) => ha.internal_id));
+  const haIdSetB = new Set(listB.map((ha) => ha.internal_id));
+  if (haIdSetA.size != haIdSetB.size) return false;
+  return haIdSetA.forEach(haIdSetB.has);
+};
 
 const onChangedSelectedHealthAuthorities: Middleware<Dispatch> = (
   store: MiddlewareAPI,
 ) => (next: Dispatch<AnyAction>) => (action: AnyAction): unknown => {
   const {
-    healthcareAuthorities: { selectedAuthorities: stateBefore },
+    healthcareAuthorities: { selectedAuthorities: haListBefore },
   } = store.getState();
 
   const result = next(action);
 
   const {
-    healthcareAuthorities: { selectedAuthorities: stateAfter },
+    healthcareAuthorities: { selectedAuthorities: haListAfter },
   } = store.getState();
 
   try {
-    if (stateBefore !== stateAfter) {
-      IntersectService.checkIntersect(stateAfter);
+    if (
+      action.type === toggleSelectedHealthcareAuthorityAction.type &&
+      !areHaListsEqual(haListBefore, haListAfter)
```

Comment:
Is the second condition redundant here? The `toggleSelectedHealthcareAuthorityAction` removes or adds a HA, so `before` and `after` lists should also be different whenever that action is dispatched.

I decided to check the state before and after because I expected that we may add more actions that could change the HA list (auto subscribe, ...) so I wanted to make this middleware more "future proof", but looking at it now, it may be less confusing and brittle to explicitly add those intersection-triggering actions to this condition.